compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -13510,6 +13510,54 @@ OMR::Z::TreeEvaluator::primitiveArraycopyEvaluator(TR::Node* node, TR::CodeGener
    byteSrcReg = cg->gprClobberEvaluate(byteSrcNode);
    byteDstReg = cg->gprClobberEvaluate(byteDstNode);
 
+   bool alreadyEvaluatedByteLenNode = false;
+
+#ifdef J9_PROJECT_SPECIFIC
+   TR::LabelSymbol* mergeLabel;
+   bool mustGenerateOOLGuardedLoadPath = cg->isConcurrentScavengeEnabled() &&
+                                         node->getArrayCopyElementType() == TR::Address;
+
+   if (mustGenerateOOLGuardedLoadPath)
+      {
+      // We must evaluate this before switching to the OOL path
```

Comment:
Perhaps explain why we need to evaluate this. You should mention that the OOL path needs the byte length for the guarded load-compress-store loop and as such it must be evaluate outside of the OOL ICF region as this register is also used in the fast path.